### ISSUE TITLE
Multi Participant Forecasting

### ISSUE THREAD

**Issue by ik362** [jdb78/pytorch-forecasting#503]
- PyTorch-Forecasting version: 0.8.5
- PyTorch version: 1.8.1
- Python version: 3.8
- Operating System: Linux

Hi there,

I had a similar question to #490 regarding how to code the group_ids for my analysis.

I am analysing time series in the context of medical data i.e. I have many time series rather than one long time series.

My time series come from two cohorts (patients and controls) and in my dataframe I have:
- a 'group_id' column where values are either ['patient', control']
- a 'subj_id' column where the values are ['subj_1', 'subj_2' ... 'subj_x']
- a 'time_idx' where the values are in the range (0, len(timeseries_subj_x))

I wanted to know if the 'subj_id' column should be part of the 'group_id' parameter or another parameter?

Thanks!

**Comment by jdb78** [comment]
Are your patients switching in and out of the control group? If not, `patient` as `group_id` should suffice. The `group_ids` identify a time series. the `time_idx` identify each datapoint in a given time series. Not sure what `subj_1` are and how it is different from the `patient`.

**Comment by ik362** [comment]
@jdb78 thanks for your help!

My ultimate question is along the lines of: can the time series in the control group be better predicted than the time series of the patients?


> Are your patients switching in and out of the control group?

In this sense the patient/control labels are static. In a statistical sense, they are _independent samples_ not _repeated measures_.

> Not sure what subj_1 are and how it is different from the patient

At the moment I have coded the data similar to this:
<img width="298" alt="Screen Shot 2021-05-16 at 1 27 58 pm" src="https://user-images.githubusercontent.com/57671726/118397161-9515c880-b64a-11eb-9fd7-cb236745b226.png">

So at the moment the `subj` column is only used as a dummy variable to identify which `time_idx` corresponds to which `subj`. I guess my question is: do I need to _"double group"_ time series into subject-level **and** group-level? Or is the subject-level implied by the `time_idx`?

**Comment by jdb78** [comment]
You can use both group levels `group` and `subj` but `subj` alone will do the job. You might want to consider using `group` as a static categorical variable on top.

Does this help?

**Comment by ik362** [comment]
Hi Jan,

Thanks for getting back to me! I have set up my TimeSeriesDataSet like this:
`training = TimeSeriesDataSet(`
`df,`
`group_ids = ['subj'],`
`target = 'source2,`
`time_idx = 'time_idx',`
`max_encoder_length = 20,`
`max_prediction_length = 20,`
`time_varying_known_reals = ['time_idx'],`
`time_varying_unknown_reals = ['source1', 'source2', 'source3', 'source4', 'source5',],`
`'static_categorical = ['group']')`

Is this set up correct?

Also, I wanted to ask about how to best compare groups? Would It make sense to use:

`predictions, x = best_tft.predict(val_dataloader, return_x=True)`
`predictions_vs_actuals = best_tft.calculate_prediction_actual_by_variable(x, predictions)`

And calculate which group had smaller differences between actuals and predictions?

Thanks for your help!

**Comment by jdb78** [comment]
Maybe not all subjects are in the validation set? I wonder if you want to include a variable for distinguish the two groups.

**Comment by ik362** [comment]
Hi Jan,

Thanks for getting back to me:

> Maybe not all subjects are in the validation set?

I used the standard procedure (from the tutorials) to define the data sets with the code: 
`validation = TimeSeriesDataSet.from_dataset(training, data, predict=True, stop_randomization=True)`
`batch_size = 16`
`train_dataloader = training.to_dataloader(train=True, batch_size=batch_size, num_workers=28)`
`val_dataloader = validation.to_dataloader(train=False, batch_size=batch_size, num_workers=28)`

>  I wonder if you want to include a variable for distinguish the two groups.

Do you mean to set `group_ids = ['subj', 'group']`?

Thanks,
Isaac
